lib/view.js

```diff
@@ -25,142 +25,127 @@
-                    var current = 0;
-                    var total = elts.length;
-                    var found = false;
-                    async.whilst(
-                        function() {
-                            return current < total;
-                        },
-                        function(callback) {
-                            var elt = elts[current++];
-                            elt.getText().then(function(txt) {
-                                if (txt === optionText) {
-                                    found = elt;
-                                }
-                                callback();
-                            });
-                        },
-                        function(err) {
-                            if (found !== false) {
-                                d.fulfill(found.click())
-                            } else {
-                                d.reject(new Error("couldn't find option with text: " + optionText));
-                            }
-                        }
-                    );
-                });
-            });
-            return d;
-        }).bind(thees);
-        context[loc + "OptionValue"] = (function(optionValue) {
-            //select an option by its text (works only for Select elements)
-            thees.drivex.find(_loc, el).then(function(selectEl) {
-                return selectEl.findElement(thees.wd.By.css("option[value='" + optionValue + "']")).click();
-            })
-        }).bind(thees);
-    },
-    resolveLocator = function(config, nemo) {
-        if (config.constructor === String) {
-            return require(nemo.props.autoBaseDir + "/locator/" + config);
-        } else if (config.locator && config.locator.constructor === Object) {
-            return config.locator;
-        } else if (config.locator && config.locator.indexOf(":") !== -1) {
-            var loco = config.locator.split(":");
-            if (loco[0] === "module") {
-                //for module style
-                return require(loco[1]);
-            } else {
-                //path style
-                return require(nemo.props.autoBaseDir + "/" + loco[1])
+var locreator = function (thees, context, loc, _loc, el) {
+    context[loc + "By"] = function () {
+      return thees.drivex.by(_loc);
+    }.bind(thees);
+    context[loc + "Present"] = function () {
+      return thees.drivex.present(_loc, el);
+    }.bind(thees);
+    context[loc] = function () {
+      return thees.drivex.find(_loc, el);
+    }.bind(thees);
+    context[loc + "Wait"] = function (timeout, msg) {
+      return thees.drivex.waitForElement(_loc, timeout || 5000, msg);
+    }.bind(thees);
+    context[loc + "WaitVisible"] = function (timeout, msg) {
+      return thees.driver.wait(function () {
+        return thees.drivex.visible(_loc).then(function (isVisible) {
+          return isVisible;
+        });
+      }, timeout || 5000, msg);
```

Comment:
The previous functionality (and documentation) of `*Wait` was to return true or false. So this builds from that except we throw an error (from `nemo-drivex.waitForElement`) when the element is not located. It wouldn't be a big change to return the web element instead of true.